### ISSUE TITLE
Start server or client based on network environment

### ISSUE THREAD

**Issue by cardonator** [deskflow/deskflow#7709]
Similar to: #7612

### Project

Deskflow

### Operating systems (OS)

- [X] Windows
- [ ] macOS
- [X] Linux
- [ ] BSD-derived
- [ ] Other

### OS versions/distros

- Windows 11
- Ubuntu 24.04

### Deskflow configuration

- Windows 11 Server
- Windows 11 Client
- Ubuntu 24.04 Server
- Ubuntu 24.04 Client

### What problem are you trying to solve?

I currently move my laptop between multiple network environments that have screens and partner devices setup in different configurations. For example, at home I have my Ubuntu 24.04 laptop set to the side of my Windows 11 machine and I use the keyboard and mouse from my Windows machine to control my laptop. At work I have a Windows 11 NUC that sits to the side of my main screen and dock and when I plug my laptop into the dock, it gets the keyboard and mouse and acts as a server that I share with the Windows machine.

### What solution do you propose?

It would be nice to have a way in Deskflow to specify what configuration should be used based on the network I'm connecting to or some other indicator. For my use case, at home I would want it to start a client and connect to my Windows machine, and at work I would want it to start as a server listening for connections from that Windows machine.

**Comment by nbolton** [comment]
Thanks for the feature request. Is this a dupe of #7612?

**Comment by rodalpho** [comment]
Rather than special-casing everything, it may make more sense (albeit sounding like a lot more work) to support more generic profiles with conditions. Profile would contain all configuration. Conditions would trigger Deskflow (or Synergy!) to load a specific profile.

So you might have a condition for your work wifi SSID, which then sets up Deskflow as you need it at work. Or you could have a trigger for the user logged-in, or IP address, or whether your laptop lid is open, or whether it has a physical mouse or external monitor attached.

**Comment by nbolton** [comment]
> support more generic profiles with conditions

Yes, that may indeed be the way to go. It'll be hard to tell what works until we start implementing stuff... but maybe we could kill two birds with one stone (both this issue and #7612).

> physical mouse or external monitor attached.

Nice idea.

**Comment by sithlord48** [comment]
Duplicate: #7612, #7889
